internal/controlplane/handlers_authz.go

```diff
@@ -392,20 +664,61 @@ func (s *Server) RemoveRole(ctx context.Context, req *minder.RemoveRoleRequest)
 		return nil, status.Errorf(codes.Internal, "error getting user: %v", err)
 	}
 
-	// Determine target project.
-	entityCtx := engine.EntityFromContext(ctx)
-	projectID := entityCtx.Project.ID
+	// Remove the existing role assignment for the user
+	as, err := s.authzClient.AssignmentsToProject(ctx, targetProject)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "error getting role assignments: %v", err)
+	}
 
-	if err := s.authzClient.Delete(ctx, identity.String(), authzrole, projectID); err != nil {
+	for _, a := range as {
+		if a.Subject == identity.String() {
+			roleToDelete, err := authz.ParseRole(a.Role)
+			if err != nil {
+				return nil, util.UserVisibleError(codes.InvalidArgument, err.Error())
```

Comment:
Just as in the other, if somehow an un parseable role is being returned by the client this points to an internal inconsistency, not an argument from the client
```suggestion
				return nil, util.UserVisibleError(codes.Internal, err.Error())
```

Reply:
Thanks! I agree on this since the respond comes from calling openfga 👍